plugins/3rdParty/qRDBIO/CMakeLists.txt

```diff
@@ -16,8 +16,17 @@ if (PLUGIN_IO_QRDB)
 	# cmake variable rdb_DIR
 	find_package(rdb REQUIRED)
 
-	# Set to include this I/O plugin in ccViewer
-	set( CC_IS_IO_PLUGIN 1 )
+	file( GLOB PLUGIN_HDR_LIST ${CMAKE_CURRENT_SOURCE_DIR}/*.h )
```

Comment:
GLOB shouldn't be used anymore (except for very specific circumstances). It doesn't allow dependencies to work properly.

You should explicitly list the sources as in the example plugin.

Reply:
That's a pain, especially for plugins with a lot of files (not this one, but I have others). I didn't understood that we lost this capacity of the previous plugin script.

Reply:
The CMake devs explicit recommend not to use GLOB for sources because it causes dependency problems.

What I did to make lists with lots of files is just select the files in the Finder and copy. That copies the file names as a list. Then paste that in and copy-paste the prefix for the line (or use an editoe with a tool to "prefix lines").

Can Windows do something similar? Or use the command line to `ls` (or equivalent) and use that list?

Reply:
Nope :|. Well of course you can always use a console and copy the list indeed.

But it means also we need to update this file anytime we change a name, or add a file. And that's super not efficient... Especially since it was working fine.

Reply:
> it means also we need to update this file anytime we change a name, or add a file

That's the whole point. Doing so maintains the dependencies properly.

> Especially since it was working fine.

For you on your setup (maybe).

At the end of the day you can do whatever you want with your plugins :-) This one (RDBIO) shouldn't even be in this repo.

Reply:
What wasn't working with the plugins before? Apart that it was not 'clean'?

qRDBIO is useful to the community, and could be moved to the 'core' plugins. It was put in '3rdParty' because it was originally developed by a '3rd party'.

Reply:
It is not "my" plugin :|

Reply:
> What wasn't working with the plugins before?

Many times I'd make a change and cmake would not re-run itself because the dependency didn't exist. So I'd get strange compile errors. Done properly, cmake is aware of the dependencies and can keep itself up to date.

> qRDBIO

Does qRDBIO not fit with qRiegl_IO, qFARO_IO, qDotProduct_IO which are not in the repo? They use proprietary libraries and are all Windows-only (I think)?

> It is not "my" plugin

;-) I was referring to the other ones you mentioned with lots of files.